packages/mongodb-chatbot-ui/src/services/conversations.ts

```diff
@@ -102,8 +122,10 @@ export class ConversationService {
   async createConversation(): Promise<Required<ConversationState>> {
     const path = `/conversations`;
     const resp = await fetch(this.getUrl(path), {
+      ...this.fetchOptions,
       method: "POST",
       headers: {
+        ...this.fetchOptions.headers,
```

Comment:
If we did accept a `Headers` object, we could use a lib like https://github.com/whitecrownclown/merge-headers for this step